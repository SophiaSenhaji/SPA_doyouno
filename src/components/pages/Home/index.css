.Animal {
    display: block;
    border: 1px solid black;
    margin-bottom: 8px;
    padding: 8px;
}

.Animal__infos {
    display: flex;
    justify-content: space-between;
}