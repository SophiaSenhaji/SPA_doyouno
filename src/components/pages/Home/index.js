import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { getAnimals } from '../../../api/animal';
import Header from '../../Header';
import SearchModal from '../../SearchModal/index';
import './index.css'
import { getDateString } from '../../../utils/date';

class Home extends Component {
    constructor() {
        super();
        this.state = {
            animals: [],
            isSearchOpened: false
        };
    }

    async componentDidMount() {
        const animals = await getAnimals();
        this.setState({
            animals
        })
    }

    onSearchClick = () => {
        this.setState({
            isSearchOpened: true
        })
    }

    onModalFound = animal => {
        const { history } = this.props;
        this.setState({
            isSearchOpened: false
        })

        history.push('/animal/' + animal._id)
    }

    onModalClose = () => {
        this.setState({
            isSearchOpened: false
        })
    }

    render() {
        const { animals, isSearchOpened } = this.state;

        return (
            <div className="Home">
                <Header withSearch withAdd onSearchClick={this.onSearchClick}/>
                {animals.map(animal => (
                    <Link className="Animal" key={animal._id} to={`/animal/${animal._id}`}>
                        <div  className="Animal__infos">
                            <div className="Animal__species">{animal.species}</div>
                            <div className="Animal__receptionDate">{getDateString(animal.receptionDate)}</div>
                        </div>
                        <div className="Animal__description">{animal.description}</div>
                    </Link>
                ))}
                {isSearchOpened && (
                    <SearchModal onFound={this.onModalFound} onClose={this.onModalClose} />
                )}
            </div>
        )
    }
}

export default Home;