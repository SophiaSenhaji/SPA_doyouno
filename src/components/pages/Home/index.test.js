import React from 'React';
import Home from './index.js';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Header component', () => {
    const animals = [
        {
            _id: 'id',
            species: 'Cat',
            receptionDate: '2019-02-05',
            description: 'description'
        },
        {
            _id: 'id',
            species: 'Dog',
            receptionDate: '2019-02-05',
            description: 'description2'
        },
    ]
    it('should render correctly', () => {
        const wrapper = shallow(<Home />);
        wrapper.setState({ animals })
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render correctly with isSearchOpened === true', () => {
        const wrapper = shallow(<Home />);
        wrapper.setState({ animals, isSearchOpened: true })
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})
