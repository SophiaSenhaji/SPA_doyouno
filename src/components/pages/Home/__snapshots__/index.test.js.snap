// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div
  className="Home"
>
  <Header
    onSearchClick={[Function]}
    withAdd={true}
    withSearch={true}
  />
  <Link
    className="Animal"
    key="id"
    replace={false}
    to="/animal/id"
  >
    <div
      className="Animal__infos"
    >
      <div
        className="Animal__species"
      >
        Cat
      </div>
      <div
        className="Animal__receptionDate"
      >
        2019-2-5
      </div>
    </div>
    <div
      className="Animal__description"
    >
      description
    </div>
  </Link>
  <Link
    className="Animal"
    key="id"
    replace={false}
    to="/animal/id"
  >
    <div
      className="Animal__infos"
    >
      <div
        className="Animal__species"
      >
        Dog
      </div>
      <div
        className="Animal__receptionDate"
      >
        2019-2-5
      </div>
    </div>
    <div
      className="Animal__description"
    >
      description2
    </div>
  </Link>
</div>
`;

exports[`Header component should render correctly with isSearchOpened === true 1`] = `
<div
  className="Home"
>
  <Header
    onSearchClick={[Function]}
    withAdd={true}
    withSearch={true}
  />
  <Link
    className="Animal"
    key="id"
    replace={false}
    to="/animal/id"
  >
    <div
      className="Animal__infos"
    >
      <div
        className="Animal__species"
      >
        Cat
      </div>
      <div
        className="Animal__receptionDate"
      >
        2019-2-5
      </div>
    </div>
    <div
      className="Animal__description"
    >
      description
    </div>
  </Link>
  <Link
    className="Animal"
    key="id"
    replace={false}
    to="/animal/id"
  >
    <div
      className="Animal__infos"
    >
      <div
        className="Animal__species"
      >
        Dog
      </div>
      <div
        className="Animal__receptionDate"
      >
        2019-2-5
      </div>
    </div>
    <div
      className="Animal__description"
    >
      description2
    </div>
  </Link>
  <SearchModal
    onClose={[Function]}
    onFound={[Function]}
  />
</div>
`;
